crates/communication/src/server/runtime.rs

```diff
@@ -68,6 +69,8 @@ where
     ) -> Result<Self, StartError> {
         let (runtime_sender, runtime_receiver) = oneshot::channel();
 
+        let parameters_changed = Arc::new(Notify::new());
+        let self_parameters_changed = parameters_changed.clone();
         let join_handle = thread::Builder::new()
             .name("communication".to_string())
             .spawn(move || {
```

Comment:
I now changed it so that the Arc Notify is created inside the thread and then sent out via the oneshot channel just like all of the other things shared between this type and the thread.
What do you think of this solution?